adapter/src/dbt/adapters/fal_experimental/support/trino.py

```diff
@@ -0,0 +1,34 @@
+from dbt.adapters.base import BaseAdapter, BaseRelation
+from dbt.adapters.base.connections import AdapterResponse
+from dbt.adapters.fal_experimental.adapter_support import new_connection
+import pandas as pd
+from dbt.adapters.sql import SQLAdapter
+from trino.dbapi import connect
+
+
+def read_relation_as_df(adapter: BaseAdapter, relation: BaseRelation) -> pd.DataFrame:
+    creds = adapter.config.credentials._db_creds
+    with connect(
+        host=creds.host,
+        port=creds.port,
+        user=creds.user,
+        catalog=creds.database,
+        schema=creds.schema,
+    ) as conn:
+        cur = conn.cursor()
+        cur.execute(f"SELECT * FROM {relation.schema}.{relation.identifier}")
+        return pd.DataFrame(cur.fetchall(), columns=[i[0] for i in cur.description])
+
+
+def write_df_to_relation(
+    adapter: SQLAdapter,
+    data: pd.DataFrame,
+    relation: BaseRelation,
+) -> AdapterResponse:
+    import sqlalchemy
+    creds = adapter.config.credentials._db_creds
+    url = f'trino://{creds.user}@{creds.host}:{creds.port}/{creds.database}'
```

Comment:
Same hre, ideally we don't build a connection in a such a manual way.

Reply:
Note `trino-python-client` has a sqlalchemy connection builder:

```
from sqlalchemy import create_engine
from trino.sqlalchemy import URL

engine = create_engine(
    URL(
        host="localhost",
        port=8080,
        catalog="system"
    ),
    connect_args={
      "session_properties": {'query_max_run_time': '1d'},
      "client_tags": ["tag1", "tag2"],
      "experimental_python_types": True,
      "roles": {"catalog1": "role1"},
    }
)
```